node/node_api.go

```diff
@@ -43,5 +43,15 @@ func (n *Node) startAPI(ctx context.Context) error {
 			log.ErrorContextE(ctx, "HTTP server stopped", err)
 		}
 	}()
+	n.APIURL = n.server.Address()
+	// Check that the server is ready before returning.
+	c, err := http.NewClient(n.APIURL)
+	if err != nil {
+		return err
+	}
+	err = c.HealthCheck(ctx)
```

Comment:
question: Is this what allowed the removal of the sleep in `updateNetworkState`?

todo: Please document why we are making this call here.

Reply:
No. I think with the changes we've made over the last few months, it's no longer necessary.

Reply:
> todo: Please document why we are making this call here.

There is already a comment for that but I can expand it a little.

Reply:
> There is already a comment for that but I can expand it a little.

Ah, my brain didn't register the comment on the constructor call as relevant for this line for some reason :)  I'm happy as-is.